docker-compose-4nodes-mps-qlight.yml

```diff
@@ -0,0 +1,788 @@
+      cp $${DDIR}/static-nodes.json $${DDIR}/permissioned-nodes.json
+      cat $${DDIR}/static-nodes.json
+      UPDATE_QUERY=".config.isMPS = false"
+      GENESIS_FILE="/examples/genesis.json"
+      CONSENSUS_RPC_API="raft"
+      if [ "${QUORUM_CONSENSUS:-istanbul}" == "istanbul" ]; then
+        CONSENSUS_RPC_API="istanbul"
+        GENESIS_FILE="/examples/istanbul-genesis.json"
+        #set the extradata for 4 nodes
+        UPDATE_QUERY=$${UPDATE_QUERY}' | .extraData = "0x0000000000000000000000000000000000000000000000000000000000000000f89af85494d8dba507e85f116b1f7e231ca8525fc9008a6966946571d97f340c8495b661a823f2c2145ca47d63c294e36cbeb565b061217930767886474e3cde903ac594f512a992f3fb749857d758ffda1330e590fa915eb8410000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0"'
+      elif [ "${QUORUM_CONSENSUS:-istanbul}" == "qbft" ]; then
+        CONSENSUS_RPC_API="qbft"
+        GENESIS_FILE="/examples/qbft-genesis.json"
+        #set the extradata for 4 nodes
+        UPDATE_QUERY=$${UPDATE_QUERY}' | .extraData = "0xf87aa00000000000000000000000000000000000000000000000000000000000000000f85494d8dba507e85f116b1f7e231ca8525fc9008a6966946571d97f340c8495b661a823f2c2145ca47d63c294e36cbeb565b061217930767886474e3cde903ac594f512a992f3fb749857d758ffda1330e590fa915ec080c0"'
+      fi
+      MPS_GENESIS_FILE=$${DDIR}/mps-genesis.json
+      echo "JQ Update Query:" $${UPDATE_QUERY}
+      jq "$${UPDATE_QUERY}" $${GENESIS_FILE} > $${MPS_GENESIS_FILE}
+      cat $${MPS_GENESIS_FILE}
+      NETWORK_ID=$$(cat $${MPS_GENESIS_FILE} | grep chainId | awk -F " " '{print $$2}' | awk -F "," '{print $$1}')
+      GETH_ARGS_raft="--raft --raftport 50400 --syncmode full"
+      GETH_ARGS_istanbul="--emitcheckpoints --istanbul.blockperiod 2 --syncmode full"
+      GETH_ARGS_qbft="--emitcheckpoints --istanbul.blockperiod 2 --syncmode full"
+      geth --datadir $${DDIR} init $${MPS_GENESIS_FILE}
+      geth \
+        --identity node$${NODE_ID}-${QUORUM_CONSENSUS:-istanbul} \
+        --datadir $${DDIR} \
+        --permissioned \
+        --nodiscover \
+        --verbosity 5 \
+        --networkid $${NETWORK_ID} \
+        --rpc \
+        --rpccorsdomain "*" \
+        --rpcvhosts "*" \
+        --rpcaddr 0.0.0.0 \
+        --rpcport 8545 \
+        --rpcapi admin,eth,debug,miner,net,shh,txpool,personal,web3,quorum,quorumExtension,$${CONSENSUS_RPC_API} \
+        --port 21000 \
+        --unlock 0 \
+        --allow-insecure-unlock \
+        --qlight.client \
+        --qlight.client.psi PS1 \
+        --qlight.client.serverNode "enode://ac6b1096ca56b9f6d004b779ae3728bf83f8e22453404cc3cef16a3d9b96608bc67c4b30db88e0a5a6c6390213f7acbe1153ff6d23ce57380104288ae19373ef@172.16.239.11:23000?discport=0&raftport=50401" \
+        --qlight.client.serverNodeRPC "http://172.16.239.11:8545/?PSI=PS1" \
```

Comment:
why not filling the PSI with the psi argument, or extracting this PSI for the client.psi?

Reply:
In the mps example there is only one qlight node (thus no need for parameters). In the multitenant example it is parameterized as you suggested.
The real reason for defining 3 qlight clients in the multitenant setup is so that each private state can be easily reached by the acceptance tests framework without having to deal with TLS or oauth tokens.